xapian-core/include/xapian/cluster.h

```diff
@@ -258,6 +258,28 @@ class XAPIAN_VISIBILITY_DEFAULT Point : public PointType {
     Document get_document() const;
 };
 
+/** Class to represent cluster centroids in the vector space
+ */
+class Centroid : public PointType {
+
+  public:
+
+    /// Constructor
+    Centroid() { magnitude = 0; }
+
+    /// This method initializes the values of a centroid to the Point 'x'
+    void set_to_point(Point &x);
+
+    /// This method divides the weight of terms in the centroid by 'size'
+    void divide(int size);
+
+    /// This method clears the terms and corresponding values of the centroid
+    void clear();
+
+    /// This method recalculates the magnitude of the centroid
+    void recalc_magnitude();
```

Comment:
Should this really be public? Maybe you need to add some more detail to the class docstring to make it clear how the `Centroid` should be used. What sort of code creates it? What sort calls these different methods? If some are done only during (eg) Cluster recalculation, then it could be protected, with a suitable `friend` declaration.